x/meshsecurity/keeper/valset_updates.go

```diff
@@ -170,11 +171,13 @@ func (k Keeper) iteratePipedValsetOperations(ctx sdk.Context, cb func(valAddress
 				return types.ErrInvalid.Wrapf("invalid slash key length %d", len(key))
 			}
 			totalSlashAmountLen := key[addrLen+2+8+8]
+			slashFractionLen := key[addrLen+2+8+8+1+totalSlashAmountLen]
 			slashInfo = &types.SlashInfo{
-				Power:            int64(sdk.BigEndianToUint64(key[addrLen+2 : addrLen+2+8])),
-				InfractionHeight: int64(sdk.BigEndianToUint64(key[addrLen+2+8 : addrLen+2+8+8])),
+				InfractionHeight: int64(sdk.BigEndianToUint64(key[addrLen+2 : addrLen+2+8])),
+				Power:            int64(sdk.BigEndianToUint64(key[addrLen+2+8 : addrLen+2+8+8])),
```

Comment:
Good catch!

This binary protocol was introduced with less fields / no slashing support. It has become increasingly complex, and mostly unmaintainable.